test/functional/test_runner.py

```diff
@@ -544,6 +544,11 @@ def run_tests(*, test_list, src_dir, build_dir, tmpdir, jobs=1, enable_coverage=
                 logging.debug("Early exiting after test failure")
                 break
 
+            shutil_disk_usage_stat_ = shutil.disk_usage(testdir)
+            if shutil_disk_usage_stat_.free < 1:
+                logging.debug("No space left on device")
+                sys.exit(-1)
```

Comment:
This will leave dangling processes

Reply:
Unable to reproduce on my system
when current test is feature_taproot.py --previous_release
when execution flow reaches the line.